### PR TITLE
Add forms for creating new posts, comments, and likes

### PR DESCRIPTION
**Pull request by shasherazi**
In this PR, I added a few rails forms with `form_with` tag.

I added the create post form on the page and `/users` page.

| Create new post button |  Create post form |
|---|---|
|image| image  |

I added the creating a new comment form on the `/users/:user_id/posts/:post_id` page under the post.

|  Create comment form |
|---|
| image  |

The like form (button) is also on the same page

| Create like form |
| --- |
|image|

I also render flashes based on error/success responses of the forms.

**Please run this command to generate the sample data easily**
```bash
rake db:populate
```

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    @comment = @post.comments.new(comment_params)
+    @comment.author = current_user
+
+    if @comment.save
+      redirect_to user_post_url(@user, @post), notice: 'Comment created successfully!'
+    else
+      flash[:errors] = @comment.errors.full_messages
+      redirect_to user_post_url(@user, @post)
+    end
+  end
```
Comment (tresorsawasawa) [inline app/controllers/comments_controller.rb]: - [x] As required for the project, kindly ensure that A comment is created on behalf of the currently logged in user (current_user) of the app. The comments controller is to have have `new` and `create` actions like you did on in the post controller.below is an example:
```ruby
class CommentsController < ApplicationController
  ...

  def new
    @comment = Comment.new
  end

  def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end
  ...
end
```

File: config/routes.rb [hunk 2]
```diff
@@ -5,6 +5,9 @@
   root 'users#index'
 
   resources :users, only: %i[index show] do
-    resources :posts, only: %i[index show]
+    resources :posts, only: %i[index show new create] do
+      resources :comments, only: %i[create]
```
Comment (tresorsawasawa) [inline config/routes.rb]: - [x] As required for the project, once you have added the `New` action with its expected code, kindly ensure that you also add its `route` in the routes.rb file. That means you should update the current routes for your comments to match the expected result.

File: app/views/posts/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<%= form_with model: [@user, @post], url: user_posts_path(@user) do |form| %>
+  <div class="form-title field">
```
Comment (tresorsawasawa) [inline app/views/posts/new.html.erb:2]: - [x] As required for the project and to make your application be completed according to the required expectations, kindly ensure there is a form in the `views/comments/new.html.erb` file which accepts input for creating comments as right it is missing. *And you should ensure that user can create a comment using the comment form you added**. Here is the error received when user wants to create a new comment:
Screenshot from 2023-06-16 06-55-00

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @shasherazi 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [x] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by shasherazi** [comment]
Thanks for the helpful review @tresorsawasawa.

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Your app meets all the changes requested by the previous reviewer. ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**